numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -28,6 +28,30 @@ extern "C" {
  * PyInt -> PyLong
  */
 
+
+/*
+ * Copy of Python private function _PyLong_AsInt This is useful in replacing
+ * PyInt_Check. This isn't strictly a Python3-Python2 compatibility function,
+ * but there is not a better place for it at this time.
+ */
+static NPY_INLINE int
+PyLong_AsInt(PyObject *obj)
```

Comment:
Is there any reason to not just use the private function? I assume it's not actually private, and just part of the "limited API" we already use elsewhere. If we put this in as is, I'd be tempted to name it `NPy_...` to avoid linker errors if this becomes public in future

Reply:
@eric-wieser Good question, I was tempted. My reason for not doing that was that it was prefixed with an underscore,  but I am not clear on what the underscore in CPython code means... It does show up in Python.h
```
#ifndef Py_LIMITED_API
PyAPI_FUNC(int) _PyLong_AsInt(PyObject *);
#endif
```
So it is not part of the limited API. Do we have a policy on that?

Reply:
AFAICT, all functions with the leading underscore are not in the limited API, but there are also some without the underscore not in the limited API, I assume for historical reasons. `_PyLong_AsInt` is not part of the API documentation.

Reply:
Unless we want to build numpy once and reuse the same binary across all supported python versions, there is little point sticking to the limited API. We are already a long way from being in that state, and i don't think there's a strong reason to start prioritizing it now.

Reply:
Agreed about using the limited API: it doesn't matter. While I am not sure we can rely on underscored versions not to be removed without warning that also is not really an issue for us...

That said, I am not sure that PyPy includes this symbol (I could not find it).  So I think copying it may just be as well (I agree, that we should use our own prefix though, I had not noticed that.

Reply:
It does not (yet) exist in PyPy. I would vote in favor of this small PR rather than depending on the CPython version.

Reply:
Pypy is a good enough reason for me, let's just use a `Npy` prefix to ensure we don't clash with cpython.